### PR TITLE
Fix for issue #10315 

### PR DESCRIPTION
**Pull request by AlexanderHefner**
jasmine-ajax stubRequest data can be a RegExp as well 

@vvakame

### CODE REVIEW

File: jasmine-ajax/jasmine-ajax.d.ts [hunk 1]
```diff
@@ -73,6 +73,9 @@ declare class MockAjax {
 
 	stubRequest(url: RegExp, data?: string, method?: string): JasmineAjaxRequestStub;
 	stubRequest(url: string, data?: string, method?: string): JasmineAjaxRequestStub;
+    
+	stubRequest(url: RegExp, data?: RegExp, method?: string): JasmineAjaxRequestStub;
+	stubRequest(url: string, data?: RegExp, method?: string): JasmineAjaxRequestStub;
```
Comment (minestarks) [inline jasmine-ajax/jasmine-ajax.d.ts]: When overload arguments only differ only by type, take advantage of union types. The 4 overloads can be condensed into:

    stubRequest(url: string | RegExp, data?: string | RegExp, method?: string): JasmineAjaxRequestStub;

Comment (AlexanderHefner) [inline jasmine-ajax/jasmine-ajax.d.ts]: I oriented myself on the already written functions, I'll change it if desire :)
